plugins/window_size/lib/src/window_size_channel.dart

```diff
@@ -113,12 +128,82 @@ class WindowSizeChannel {
   void setWindowFrame(Rect frame) async {
     assert(!frame.isEmpty, 'Cannot set window frame to an empty rect.');
     assert(frame.isFinite, 'Cannot set window frame to a non-finite rect.');
-    try {
-      await _platformChannel.invokeMethod(_setWindowFrameMethod,
-          [frame.left, frame.top, frame.width, frame.height]);
-    } on PlatformException catch (e) {
-      print('Platform exception setting window frame: ${e.message}');
+    await _platformChannel.invokeMethod(_setWindowFrameMethod,
+        [frame.left, frame.top, frame.width, frame.height]);
+  }
+
+  // Window minimum size unconstrained is passed over the channel as -1.
+  double _setWindowMinSizeUnconstrainedCheck(double size) {
+    if (size > 0) {
+      return size;
     }
+    return -1;
+  }
+
+  /// Sets the minimum size of the window containing this Flutter instance.
+  void setWindowMinSize(Size size) async {
+    await _platformChannel.invokeMethod(_setWindowMinimumSizeMethod, [
+      _setWindowMinSizeUnconstrainedCheck(size.width),
+      _setWindowMinSizeUnconstrainedCheck(size.height),
+    ]);
+  }
+
+  // Window maximum size unconstrained is passed over the channel as -1.
+  double _setWindowMaxSizeUnconstrainedCheck(double size) {
+    if (size != double.infinity) {
+      return size;
+    }
+    return -1;
+  }
+
+  /// Sets the maximum size of the window containing this Flutter instance.
+  void setWindowMaxSize(Size size) async {
+    await _platformChannel.invokeMethod(_setWindowMaximumSizeMethod, [
+      _setWindowMaxSizeUnconstrainedCheck(size.width),
+      _setWindowMaxSizeUnconstrainedCheck(size.height),
+    ]);
+  }
+
+  // Window minimum size unconstrained is passed over the channel as -1.
+  double _getWindowMinSizeUnconstrainedCheck(double size) {
+    if (size != -1) {
+      return size;
+    }
+    return 0;
+  }
+
+  /// Gets the minimum size of the window containing this Flutter instance.
+  Future<Size> getWindowMinSize() async {
+    final response =
+        await _platformChannel.invokeMethod(_getWindowMinimumSizeMethod);
+    return _sizeFromWHList(
+      List<double>.from(
+        response[_windowSizeKey]
+            .cast<double>()
+            .map(_getWindowMinSizeUnconstrainedCheck),
+      ),
+    );
+  }
+
+  // Window maximum size unconstrained is passed over the channel as -1.
+  double _getWindowMaxSizeUnconstrainedCheck(double size) {
```

Comment:
Essentially the same comment here; `_maxDimensionFromChannelRepresentation` would be clearer IMO.